server/routers/authenticate/authProviders/passkeys.ts

```diff
@@ -0,0 +1,714 @@
+        });
+        
+        if (!userProfile) {
+          throw new TRPCError({
+            code: "NOT_FOUND",
+            message: "User not registered. Please register with another authentication method first.",
+          });
+        }
+        
+        // Store the email temporarily
+        await ctx.prisma.passkeyChallenge.create({
+          data: {
+            userId: tempUserId,
+            value: email,
+            version: "email-verification",
+            createdAt: new Date(),
+          },
+        });
+        
+        // Generate registration options
+        const options = await generateRegistrationOptions({
+          rpName: relyingPartyName,
+          rpID: relyingPartyID,
+          userID: stringToUint8Array(tempUserId),
+          userName: email,
+          attestationType: "direct",
+          authenticatorSelection: {
+            residentKey: "preferred",
+            userVerification: "preferred",
+            authenticatorAttachment: "platform",
+          },
+        });
+        
+        // Store the challenge
+        const challenge = options.challenge;
+        await ctx.prisma.passkeyChallenge.create({
+          data: {
+            userId: tempUserId,
+            value: challenge,
+            version: "1",
+            createdAt: new Date(),
+          },
+        });
+        
+        return {
+          options,
+          tempUserId,
+        };
+      }
+      
+      // Generate registration options
+      const options = await generateRegistrationOptions({
+        rpName: relyingPartyName,
+        rpID: relyingPartyID,
+        userID: stringToUint8Array(tempUserId),
+        userName: tempUserId, // Will be updated later by the user
+        attestationType: "direct",
+        authenticatorSelection: {
+          residentKey: "preferred",
+          userVerification: "preferred",
+          authenticatorAttachment: "platform",
+        },
+      });
+
+      // Store the challenge in the database
+      const challenge = options.challenge;
+      await ctx.prisma.passkeyChallenge.create({
+        data: {
+          userId: tempUserId,
+          value: challenge,
+          version: "1",
+          createdAt: new Date(),
+        },
+      });
+
+      // Return both the options and the temporary user ID
+      return {
+        options,
+        tempUserId,
+        requiresOtp: false,
+      };
+    }),
+
+  verifyOtp: publicProcedure
+    .input(
+      z.object({
+        verificationId: z.string(),
+        otp: z.string(),
+        email: z.string().email(),
+      })
+    )
+    .mutation(async ({ input, ctx }) => {
+      const { verificationId, otp, email } = input;
+      const supabase = await createServerClient();
+      
+      try {
+        // Get the credential data
+        const credentialRecord = await ctx.prisma.passkeyChallenge.findFirst({
+          where: {
+            userId: verificationId,
+            version: "credential-verification",
```

Comment:
I cannot find where these `credential-verification` challenges are created, not sure if it's just the quick search not finding it on the page :S